### PR TITLE
Issue #9530: Update javadoc variable check test to use unique input file in each test method

### LINKED ISSUE #9530: update JavadocVariableCheckTest to use unique Input file in each test method
JavadocVariableCheckTest.java file reuse some Input files few time in few different test method.
Expected: each test method in Test class use it own Input file and do not share it with any other test method.

more details - https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-create-Input-files-for-tests

### PR DESCRIPTION
**Pull request by hiron1999**
Fixes #9530
Made some minor code changes in existing unit tests, added unique input files for each test method

### CODE REVIEW

File: config/checkstyle_input_suppressions.xml [hunk 1]
```diff
@@ -1292,12 +1292,24 @@
...
   <suppress id="ConfigCommentOnTopInputs"
             files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableNoJavadoc.java"/>
+  <suppress id="ConfigCommentOnTopInputs"
+            files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableNoJavadoc2.java"/>
+  <suppress id="ConfigCommentOnTopInputs"
+            files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableNoJavadoc3.java"/>
+  <suppress id="ConfigCommentOnTopInputs"
+            files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableNoJavadoc4.java"/>
+  <suppress id="ConfigCommentOnTopInputs"
+            files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableNoJavadoc5.java"/>
```
Comment (AkMo3) [inline config/checkstyle_input_suppressions.xml:1306]: @hiron1999 Please do not add suppressions, and try to resolve as many previous suppressions associated with this test.

**Review by AkMo3 — CHANGES_REQUESTED** [review]
Point:

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly.java [hunk 2]
```diff
@@ -3,17 +3,21 @@
 // Created: 2001
 ////////////////////////////////////////////////////////////////////////////////
 package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;
-
+/* Config:
+ * scope = private
+ *
+ *
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly.java:9]: Remove these extra lines at 8, 9.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly2.java [hunk 3]
```diff
@@ -0,0 +1,121 @@
+
+
+
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;
+/* Config:
+ * scope = public
+ *
+ *
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly2.java:9]: Same here.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;
+
+/**
+ * Tests having inner types
+ * @author Oliver Burn
+ **/
+/* Config:
+ * scope = public
+ * excludeScope = null
+ *
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java:14]: Same as above.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java [hunk 5]
```diff
@@ -8,26 +8,31 @@
  * Tests having inner types
  * @author Oliver Burn
  **/
+/* Config:
+ * scope = private
+ * excludeScope =null
+ *
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java:14]: Can be rewritten to some thing 
```
/**
 * Config:
 * scope = private
 * excludeScope =null
 * 
 * Tests having inner types
 * @author Oliver Burn
 **/
```

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;// violation
+/* Config:
+ * scope = protected
+ * excludeScope = null
+ *
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java:5]: Remove extra line.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc3.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;// violation
+/* Config:
+ * scope = private
+ * excludeScope = protected
+ *
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc3.java:5]: Same as above.

File: config/checkstyle_input_suppressions.xml [hunk 8]
```diff
@@ -1292,12 +1292,15 @@
...
   <suppress id="ConfigCommentOnTopInputs"
             files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableNoJavadoc.java"/>
+
   <suppress id="ConfigCommentOnTopInputs"
             files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableNoJavadocNeededInLambda.java"/>
   <suppress id="ConfigCommentOnTopInputs"
             files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariablePublicOnly.java"/>
+
   <suppress id="ConfigCommentOnTopInputs"
             files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableTags.java"/>
```
Comment (AkMo3) [inline config/checkstyle_input_suppressions.xml:1305]: This should be removed. As you had updated them properly now. Similarly remove them for style validation also.

**Review by AkMo3 — CHANGES_REQUESTED** [review]
Some points:

File: config/checkstyle_input_suppressions.xml (hunk 1 above)
Comment (AkMo3) [reply config/checkstyle_input_suppressions.xml]: You might have created this branch directly from your last commited branch, so this branch has been cloned similar to previous one.

File: config/checkstyle_input_suppressions.xml (hunk 8 above)
Comment (hiron1999) [reply config/checkstyle_input_suppressions.xml:1305]: yes , I am working on it

**Comment by hiron1999** [comment]
@AkMo3  please check if the changes are fine

**Comment by AkMo3** [comment]
@hiron1999 Please squash commits into one. This would also enable me to see all the changes at once.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java (hunk 5 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java:11]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java (hunk 4 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java:11]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly.java (hunk 2 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly2.java (hunk 3 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly2.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java (hunk 6 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc3.java (hunk 7 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc3.java]: Resolved.

File: config/checkstyle_input_suppressions.xml (hunk 8 above)
Comment (AkMo3) [reply config/checkstyle_input_suppressions.xml]: Resolved.

**Comment by AkMo3** [comment]
@hiron1999 Please resolve the above reviews from your side. After that there are some new observations, but reviewing before resolving above ones might lead to double occurrence of previous ones. Also, still there are two commits, please squash them into one.

File: config/checkstyle_input_suppressions.xml (hunk 1 above)
Comment (AkMo3) [reply config/checkstyle_input_suppressions.xml]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly.java (hunk 2 above)
Comment (hiron1999) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly.java]: resolved

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java [hunk 9]
```diff
@@ -0,0 +1,84 @@
+// violation
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java:1]: Why violation here? I don't see any violations in this test
https://github.com/checkstyle/checkstyle/blob/dabdffd3716c6fdbb780faff63e26ceb08d2bbe3/src/test/java/com/puppycrawl/tools/checkstyle/checks/javadoc/JavadocVariableCheckTest.java#L91-L99

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+
+
+
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java:4]: Please remove these empty lines.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java [hunk 11]
```diff
@@ -5,29 +5,34 @@
...
- **/
+ */
 class InputJavadocVariableInner
 {
     // Ignore - two violations
     class InnerInner2
     {
         // Ignore
-        public int fData;
+        public int fData;   // violation
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java:22]: Keep only a single space between `;` and `//` as it has been followed in rest of the PR.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java [hunk 12]
```diff
@@ -5,29 +5,34 @@
...
     }
 
     // Ignore - 2 violations
     interface InnerInterface2
     {
         // Ignore - should be all upper case
-        String data = "zxzc";
+        String data = "zxzc";   // violation
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java:29]: Same as above.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java [hunk 13]
```diff
@@ -5,29 +5,34 @@
...
 
         // Ignore
         class InnerInterfaceInnerClass
         {
             // Ignore - need Javadoc and made private
-            public int rData;
+            public int rData;   // violation
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java:35]: Same as above.

File: config/checkstyle_input_suppressions.xml [hunk 14]
```diff
@@ -1290,16 +1290,7 @@
...
-            files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableInner.java"/>
-  <suppress id="ConfigCommentOnTopInputs"
-            files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableNoJavadoc.java"/>
-  <suppress id="ConfigCommentOnTopInputs"
-            files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableNoJavadocNeededInLambda.java"/>
-  <suppress id="ConfigCommentOnTopInputs"
-            files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariablePublicOnly.java"/>
-  <suppress id="ConfigCommentOnTopInputs"
-            files="javadoc[\\/]javadocvariable[\\/]InputJavadocVariableTags.java"/>
+
```
Comment (AkMo3) [inline config/checkstyle_input_suppressions.xml:1293]: Please remove this empty line.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java [hunk 15]
```diff
@@ -0,0 +1,124 @@
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;// violation
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java:1]: Space before `//` and empty line after package declaration. This mistake is in lot of files. Please go through them and correct them all.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc.java [hunk 16]
```diff
@@ -1,22 +1,26 @@
-package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;
-
+package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;// violation
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc.java:1]: Space before `//` and empty line after package declaration.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableTags.java [hunk 17]
```diff
@@ -301,14 +306,14 @@ public int foo(Object _arg) {
 
 enum InputJavadocVariableTagsEnum
 {
-    CONSTANT_A,
+    CONSTANT_A,// violation
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableTags.java:309]: Space before `//`

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableTags.java [hunk 18]
```diff
@@ -4,11 +4,16 @@
 ////////////////////////////////////////////////////////////////////////////////
 package com.puppycrawl.tools.checkstyle.checks.javadoc.javadocvariable;
 import java.io.IOException;
-// Tests for Javadoc tags.
+/**
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableTags.java:7]: Empty line after import.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly2.java [hunk 19]
```diff
@@ -0,0 +1,121 @@
+
+
+
+
```
Comment (AkMo3) [inline src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly2.java:4]: Please remove these empty lines.

**Review by AkMo3 — CHANGES_REQUESTED** [review]
Points:

File: config/checkstyle_input_suppressions.xml (hunk 1 above)
Comment (AkMo3) [reply config/checkstyle_input_suppressions.xml]: @hiron1999 Please resolve this one. Also, Please comment `Done` to each suggested changes so as to keep track which ones have been done.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java (hunk 9 above)
Comment (hiron1999) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java:1]: I have same question , but if i try to remove this  it gives error on ```mvn clean verify```
error massage:
```
[ERROR] [checkstyle] [ERROR] H:\GSOC\checkstyle\src\test\resources\com\puppycrawl\tools\checkstyle\checks\javadoc\javadocvariable\InputJavadocVariableInner2.java:0: All check input files in the 'src/(it|test)/resources*/' folder must                    have trailing comments "// ok" and/or "// violation". [StyleValidationCommentInInputs]
```

Comment (hiron1999) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java:1]: so if i replace ``` // validation ``` with ```// ok``` is that fine?

Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java:1]: No. At top of file, there must only be package declaration. There are so many `ok` cases below. Just comment in some (or one) of the,/

Comment (hiron1999) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java:1]: ok

**Comment by hiron1999** [comment]
@AkMo3  please check the  suggested changes

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java (hunk 9 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java (hunk 11 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java (hunk 12 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java (hunk 13 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner.java]: Resolved.

File: config/checkstyle_input_suppressions.xml (hunk 14 above)
Comment (AkMo3) [reply config/checkstyle_input_suppressions.xml]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java (hunk 15 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java]: Not done. [InputJavadocVariableNoJavadoc2.java](https://github.com/checkstyle/checkstyle/blob/fb62f8f570707d313d82ebe22d65f209f6892a2e/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java) and InputJavadocVariableNoJavadoc4.java are missed.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc.java (hunk 16 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableTags.java (hunk 17 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableTags.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly2.java (hunk 19 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariablePublicOnly2.java]: Resolved.

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java (hunk 10 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableInner2.java]: Resolved.

**Comment by AkMo3** [comment]
@hiron1999 Please see [#9702 Comment](https://github.com/checkstyle/checkstyle/pull/9702#discussion_r600436609), only it remains unsolved. Reset are resolved. Great work.

**Comment by hiron1999** [comment]
> @hiron1999 Please see [#9702 Comment](https://github.com/checkstyle/checkstyle/pull/9702#discussion_r600436609), only it remains unsolved. Reset are resolved. Great work.

@AkMo3  yes i have missed that one , now it is resolved

**Review by AkMo3 — APPROVED** [review]

File: src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java (hunk 15 above)
Comment (AkMo3) [reply src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/javadocvariable/InputJavadocVariableNoJavadoc2.java]: Resolved.

**Pull request merged**